classes/WP_CLI/ActionScheduler_WPCLI_System_Command.php

```diff
@@ -0,0 +1,213 @@
+	 * @param array $args       Positional args.
+	 * @param array $assoc_args Keyed args.
+	 * @uses $this->get_current_datastore()
+	 * @return void
+	 *
+	 * @subcommand data-store
+	 */
+	public function datastore( array $args, array $assoc_args ) {
+		echo $this->get_current_datastore();
+	}
+
+	/**
+	 * Print in-use runner class.
+	 *
+	 * @param array $args       Positional args.
+	 * @param array $assoc_args Keyed args.
+	 * @uses $this->get_current_runner()
+	 * @return void
+	 */
+	public function runner( array $args, array $assoc_args ) {
+		echo $this->get_current_runner();
+	}
+
+	/**
+	 * Get system status.
+	 *
+	 * @param array $args       Positional args.
+	 * @param array $assoc_args Keyed args.
+	 * @uses $this->get_current_datastore()
+	 * @uses $this->get_latest_version()
+	 * @uses $this->print_statuses()
+	 * @return void
+	 */
+	public function status( array $args, array $assoc_args ) {
+		/**
+		 * Get runner status.
+		 *
+		 * @link https://github.com/woocommerce/action-scheduler-disable-default-runner
+		 */
+		$runner_enabled = has_action( 'action_scheduler_run_queue', array( ActionScheduler::runner(), 'run' ) );
+
+		\WP_CLI::line( sprintf( 'Data store: %s', $this->get_current_datastore() ) );
+		\WP_CLI::line( sprintf( 'Runner: %s%s', $this->get_current_runner(), ( $runner_enabled ? '' : ' (disabled)' ) ) );
+		\WP_CLI::line( sprintf( 'Version: %s', $this->get_latest_version() ) );
+
+		$rows              = array();
+		$action_counts     = $this->store->action_counts();
+		$oldest_and_newest = $this->get_oldest_and_newest( array_keys( $action_counts ) );
+
+		foreach ( $action_counts as $status => $count ) {
+			$rows[] = array(
+				'status' => $status,
+				'count'  => $count,
+				'oldest' => $oldest_and_newest[ $status ]['oldest'],
+				'newest' => $oldest_and_newest[ $status ]['newest'],
+			);
+		}
+
+		$formatter = new \WP_CLI\Formatter( $assoc_args, array( 'status', 'count', 'oldest', 'newest' ) );
+		$formatter->display_items( $rows );
+	}
+
+	/**
+	 * Get latest or all system versions.
+	 *
+	 * ## OPTIONS
+	 *
+	 * [--all]
+	 * : Get all system versions.
+	 *
+	 * @param array $args       Positional args.
+	 * @param array $assoc_args Keyed args.
+	 * @uses \ActionScheduler_Versions::get_versions()
+	 * @uses \WP_CLI\Formatter::display_items()
+	 * @uses $this->get_latest_version()
+	 * @return void
+	 */
+	public function version( array $args, array $assoc_args ) {
+		$all      = (bool) get_flag_value( $assoc_args, 'all' );
+		$instance = \ActionScheduler_Versions::instance();
+
+		if ( $all ) {
+			$versions = $instance->get_versions();
+
+			$rows = array();
+
+			foreach ( $versions as $version => $callback ) {
+				$rows[ $version ] = array(
+					'version'  => $version,
+					'callback' => $callback,
+				);
+			}
+
+			uksort( $rows, 'version_compare' );
+
+			$formatter = new \WP_CLI\Formatter( $assoc_args, array( 'version', 'callback' ) );
+			$formatter->display_items( $rows );
+
+			return;
+		}
```

Comment:
Currently the output is something like this:

```
+---------+-------------------------------------------+
| version | callback                                  |
+---------+-------------------------------------------+
| 3.6.0   | action_scheduler_initialize_3_dot_6_dot_0 |
| 3.8.1   | action_scheduler_initialize_3_dot_8_dot_1 |
+---------+-------------------------------------------+
```

Could it be useful to indicate which is active/loaded?

```
+---------+-------------------------------------------+--------+
| version | callback                                  | active |
+---------+-------------------------------------------+--------+
| 3.6.0   | action_scheduler_initialize_3_dot_6_dot_0 | no     |
| 3.8.1   | action_scheduler_initialize_3_dot_8_dot_1 | yes    |
+---------+-------------------------------------------+--------+
```

We generally hope it will be the latest version, but that is not always so (and my assumption is that this sub-command exists to aid debugging).

Reply:
Oh whoa. What would the situation be where the latest isn't used?

Reply:
If several versions of Action Scheduler are present and one is 3.2.0 or earlier then, depending on how it is initialized, it may take precedence over later versions [(related)](https://developer.woocommerce.com/2021/10/12/best-practices-for-deconflicting-different-versions-of-action-scheduler/). Not just hypothetical, we came across one such situation quite recently, impacting WooCommerce.

Reply:
That said, the proposed change could be left until a future iteration.

Reply:
Interesting. The `active` column was added.